kotsadm/pkg/automation/automation.go

```diff
@@ -84,48 +92,150 @@ func AutomateInstall() error {
 		// check license expiration
 		expired, err := kotspull.LicenseIsExpired(verifiedLicense)
 		if err != nil {
-			logger.Error(err)
+			logger.Error(errors.Wrap(err, "failed to check is lincense is expired"))
 			continue
 		}
 		if expired {
-			logger.Error(errors.New("license is expired"))
+			logger.Error(errors.Errorf("license is expired for app %s", verifiedLicense.Spec.AppSlug))
 			continue
 		}
 
 		skipImagePush, err := kotsutil.IsImagesPushedSet(kotsadmtypes.KotsadmConfigMap)
 		if err != nil {
-			return errors.Wrap(err, "failed to get existing kotsadm config map")
+			logger.Error(errors.Wrap(err, "failed to get existing kotsadm config map"))
+			continue
 		}
 
 		desiredAppName := strings.Replace(verifiedLicense.Spec.AppSlug, "-", " ", 0)
 		upstreamURI := fmt.Sprintf("replicated://%s", verifiedLicense.Spec.AppSlug)
 
 		a, err := store.GetStore().CreateApp(desiredAppName, upstreamURI, string(license), verifiedLicense.Spec.IsAirgapSupported, skipImagePush)
 		if err != nil {
-			logger.Error(err)
+			logger.Error(errors.Wrap(err, "failed to create app record"))
 			continue
 		}
 
-		// complete the install online
-		pendingApp := installationtypes.PendingApp{
-			ID:          a.ID,
-			Slug:        a.Slug,
-			Name:        a.Name,
-			LicenseData: string(license),
-		}
-		_, err = online.CreateAppFromOnline(&pendingApp, upstreamURI, true)
+		airgapData, err := getAirgapData(clientset, verifiedLicense)
 		if err != nil {
-			logger.Error(err)
+			logger.Error(errors.Wrapf(err, "failed to load airgap data for %s", verifiedLicense.Spec.AppSlug))
 			continue
 		}
 
+		if skipImagePush && airgapData != nil {
```

Comment:
The logic here .. i think it's right, but i think a comment above explaining these `if` blocks will be really useful for future-us.